internal/blocksync/pool.go

```diff
@@ -130,27 +130,23 @@ func (*BlockPool) OnStop() {}
 
 // spawns requesters as needed
 func (pool *BlockPool) makeRequestersRoutine(ctx context.Context) {
-	for {
-		if !pool.IsRunning() {
-			break
+	for pool.IsRunning() {
+		if ctx.Err() != nil {
+			return
 		}
 
 		_, numPending, lenRequesters := pool.GetStatus()
-		switch {
-		case numPending >= maxPendingRequests:
-			// sleep for a bit.
-			time.Sleep(requestIntervalMS * time.Millisecond)
-			// check for timed out peers
-			pool.removeTimedoutPeers()
-		case lenRequesters >= maxTotalRequesters:
-			// sleep for a bit.
-			time.Sleep(requestIntervalMS * time.Millisecond)
-			// check for timed out peers
+		if numPending >= maxPendingRequests || lenRequesters >= maxTotalRequesters {
+			// This is preferable to using a timer because the request interval
+			// is so small. Larger request intervals may necessitate using a
+			// timer/ticker.
+			time.Sleep(requestInterval)
 			pool.removeTimedoutPeers()
-		default:
-			// request for more blocks.
-			pool.makeNextRequester(ctx)
+			continue
 		}
+
+		// request for more blocks.
+		pool.makeNextRequester(ctx)
```

Comment:
doesn't this end up spinning pretty fast if the condition on 139 isn't hit? I guess that's the same as its always been.

Reply:
Yes, I think it's in order to quickly spin up a bunch of requesters up to the relevant limits (`maxPendingRequests` and `maxTotalRequesters`) and then go into the routine that cleans up timed-out peers. But I'm not totally familiar with this block sync code.

Reply:
Yeah I noticed that too, but this hasn't changed the existing behaviour. Because of how the Go scheduler works, spinning is almost always a bad idea. Even a minimal delay is advisable (but I didn't want to go down the rabbit hole for this change). I might take a pass for this pattern sometime—it's likely at the heart of some of our contention throttling issues.